### ISSUE TITLE
Creating a SUVIMap from SUVI L1b/L2 files is broken in SunPy 2.1.1

### ISSUE THREAD

**Issue by cbethge** [sunpy/sunpy#5043]
### Description

The sunpy.map.sources.SUVIMap class expects data/header input from astropy.io.fits. Probably due to changes in WCS-related modules from SunPy 2.0.1 -> 2.1.1, creating a SUVIMap from SUVI L1b/L2 files is no longer working. Metadata in the headers seems to be missing to make it work, but I can't tell which additional metadata is required now.

### Expected vs actual behavior

**Expected behaviour:** when creating a SUVIMap from data/header inputs in SunPy 2.0.1, it would throw an AstropyDeprecationWarning ("WARNING: AstropyDeprecationWarning: The private astropy._erfa module has been made into its own package, pyerfa, which is a dependency of astropy and can be imported directly using "import erfa" [astropy._erfa]"), but it would still create it.

**Actual behaviour in SunPy 2.1.1:** Quits with "TypeError: Can not create a MetaDict from this type input".

### Steps to Reproduce

```
In [1]: from astropy.io import fits
   ...: from sunpy.map.sources import SUVIMap
   ...: hdu = fits.open('dr_suvi-l2-ci195_g16_s20200220T180000Z_e20200220T180400Z_v1-0-0.fits')
   ...: suvimap = SUVIMap(hdu[1].data, hdu[1].header)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-777075e1d3d8> in <module>
      2 from sunpy.map.sources import SUVIMap
      3 hdu = fits.open('dr_suvi-l2-ci195_g16_s20200220T180000Z_e20200220T180400Z_v1-0-0.fits')
----> 4 suvimap = SUVIMap(hdu[1].data, hdu[1].header)

~/opt/anaconda3/lib/python3.7/site-packages/sunpy/map/sources/suvi.py in __init__(self, data, header, **kwargs)
     72     def __init__(self, data, header, **kwargs):
     73 
---> 74         super().__init__(data, header, **kwargs)
     75 
     76         # Fill in some missing info

~/opt/anaconda3/lib/python3.7/site-packages/sunpy/map/mapbase.py in __init__(self, data, header, plot_settings, **kwargs)
    187                           "Data will be truncated to the first two dimensions.", SunpyUserWarning)
    188 
--> 189         super().__init__(data, meta=MetaDict(header), **kwargs)
    190 
    191         # Correct possibly missing meta keywords

~/opt/anaconda3/lib/python3.7/site-packages/sunpy/util/metadata.py in __init__(self, *args)
     38                 items = adict.items()
     39             else:
---> 40                 raise TypeError("Can not create a MetaDict from this type input")
     41 
     42             self._check_str_keys(items)

TypeError: Can not create a MetaDict from this type input
```

### System Details

- SunPy Version: 2.1.1
- Astropy Version: 4.2
- Python Version: 3.7.9
- OS information: Mac OS X 10.16 (i386)

**Comment by Cadair** [comment]
Hi @cbethge Thanks for the bug report. Do you have a link to that file to make debugging a little faster?

**Comment by cbethge** [comment]
Hi @Cadair! Sure, thanks for looking at the issue. Any of the files in the subfolder in

https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes16/l2/data/suvi-l2-ci171/ 

should work, so for example

https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes16/l2/data/suvi-l2-ci171/2021/02/28/dr_suvi-l2-ci171_g16_s20210228T000000Z_e20210228T000400Z_v1-0-1.fits

Fyi, the SUVI files do have some header issues in the L1b files (that are being taken into account in SUVIMap), but there should not be any issues regarding the WCS compliance. So my guess would be that SUVI is not the only instrument affected by this problem.

**Comment by hayesla** [comment]
Hey @cbethge and @Cadair 

I had a look at this and seems to work if you pass the file to `sunpy.map.Map`

```python
import sunpy.map
>>> my_map = sunpy.map.Map("dr_suvi-l2-ci195_g16_s20200220T180000Z_e20200220T180400Z_v1-0-0.fits")
>>> print(type(my_map))
<class 'sunpy.map.sources.suvi.SUVIMap'>

```

I'm not too sure why it's having issues with the `SUVIMap`

**Comment by Cadair** [comment]
You might need to convert the astropy Header object to a dict before you pass it to Metadict via the map constructor.

That being said, direct instantiation of a map subclass isn't really the supported route, we recommend you use the `Map` factory.

**Comment by cbethge** [comment]
Thanks @hayesla, that's good to know! I would prefer a "regular" Sunpy map anyway, because I can pass a list of files and don't have to use data/header inputs. However, that won't work for the L1b files unfortunately, because of the SUVI header issues (they use the CONTINUE keyword incorrectly). For the L2 files, that should be fine though.

**Comment by Cadair** [comment]
You can pass in a `(data, header)` pair to `Map` instead of a filename if that helps.

**Comment by cbethge** [comment]
Ok, thanks for the replies! I guess my two options are: 

- using sunpy.map.Map for the L2 files
- reading in L1b files as data/header, fixing the CONTINUE issue, and then passing it to sunpy.map.Map as data/header input

I wasn't aware that the instrument classes were not supposed to be called directly. I don't know if that means that this issue (if it is one) still needs to be fixed.

**Comment by hayesla** [comment]
Yes so, you can do:

* Pass file directly to `sunpy.map.Map`:
``` python
>>> my_map = sunpy.map.Map(filename)
```

* pass the data and header from fits file to `sunpy.map.Map`
``` python
>>> hdu = fits.open(filename)
>>> my_map = sunpy.map.Map(hdu[1].data, hdu[1].header)
```

* convert the header to a dictionary and then pass to `SUVIMap` as @Cadair mentioned rectifies the issue you were having:
```python
>>> from collections import OrderedDict
>>> hdu = fits.open(filename)
>>> my_map = SUVIMap(hdu[1].data, OrderedDict(hdu[1].header))
```

However, the best practice as @Cadair  says is to use `sunpy.map.Map`. I think the fact that a dictionary can be passed means that this is no longer an issue with the SUVI files and map. 

:)

**Comment by cbethge** [comment]
Agreed, just an issue with my understanding of how things are supposed to be used :-) I'll close this then, thanks for the help!

**Comment by dstansby** [comment]
On the original issue, it bisects to https://github.com/sunpy/sunpy/commit/584aa55736acc98880424512fe6880e7de04a4ea - since I broke this (sorry!) and it's a regression, I will try and write a bugfix.

**Comment by nabobalis** [comment]
Should we convert from a dict to metadict?

**Comment by dstansby** [comment]
I've got a fix, just writing a regression test.

**Comment by nabobalis** [comment]
This is now fixed in the newly released sunpy 2.1.2
